### PR TITLE
Retry lookups with larger buffer sizes on ERANGE

### PR DESCRIPTION
**Pull request by Freaky**
This adds a simple resize loop to lookups on `ERANGE`.

### CODE REVIEW

File: src/base.rs [hunk 1]
```diff
@@ -322,14 +322,22 @@ unsafe fn members(groups: *mut *mut c_char) -> Vec<OsString> {
...
+    loop {
+        let r = unsafe {
+            libc::getpwuid_r(uid, &mut passwd, buf.as_mut_ptr(), buf.len(), &mut result)
+        };
+
+        if r != libc::ERANGE {
+            break;
+        }
+
+        buf.resize(buf.len() * 2, 0);
```
Comment (max-heller) [inline src/base.rs]: These lines are causing borrow checker errors on 1.31. Defining the new length on the line above should help the old compiler out.

Comment (Freaky) [reply src/base.rs]: Ta for the reminder, fixed.

**Comment by max-heller** [comment]
@ogham anything you need changed here, or can it be merged?

**Comment by ogham** [comment]
This is great! Thanks.

**Comment by ogham** [comment]
Released as [v0.11.0](https://github.com/ogham/rust-users/releases/tag/v0.11.0).

**Pull request merged**